### ISSUE TITLE
Unexpected (and very bad) data loss

### ISSUE THREAD

**Issue by theodoreb** [yarnpkg/yarn#1224]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I run `yarn` in my project folder and it tries to delete everything from the **parent** directory. See console output: 
```sh
theodore@monkey:~/DEV/PERSO/PEP$ yarn
yarn install v0.15.1
info No lockfile found.
[1/4] Resolving packages...
warning intern > istanbul > fileset > minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
[2/4] Fetching packages...
[3/4] Linking dependencies...
error EACCES: permission denied, unlink '~/DEV/PERSO/pwa/docroot/sites/default/default.settings.php'
    at Error (native)
info Visit http://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
Tried with latest yarn version, same thing.

**If the current behavior is a bug, please provide the steps to reproduce.**
1. Checkout  the PEP repository: https://github.com/jquery/PEP
2. Run `yarn`
3. Watch your parent folder get deleted.

**What is the expected behavior?**
My data is not deleted.

**Please mention your node.js, yarn and operating system version.**
See redacted [yarn-error.txt](https://github.com/yarnpkg/yarn/files/537877/yarn-error.txt).

**Comment by sevenseacat** [comment]
Verified with the PEP repository on Yarn v0.16.1, and node v6.9.1. All contents of the parent folder (files and folders) deleted. `npm install` doesn't have the same behaviour. 

However, I've done the same thing with other projects and this bug hasn't happened...

**Comment by chlunde** [comment]
Looks like it was fixed by #1441
